### PR TITLE
Advanced-Building-Blocks--Enumerables

### PR DESCRIPTION
**Pull request by Davidosky007**
# Advanced-Building-Blocks--Enumerable
in this project we
- we used the module keyword
- Creating enumerable methods accepting a block using yield() and #call

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+
+    counter = 0
+    while counter < arr.length
+      yield(arr[counter])
+      counter += 1
+    end
+    arr
+  end
```
Comment (mahmoud717) [inline main.rb]: - my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self
- my_each when a block is given when self is a hash behaves like #each calls the given block once for each element in self

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    arr = self if instance_of?(Array)
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+    counter = 0
+    while counter < arr.length
+      yield(arr[counter], counter)
+      counter += 1
+    end
+    arr
+  end
```
Comment (mahmoud717) [inline main.rb]: - my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self
- my_each_with_index when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+      my_each { |v| return false if v.nil? || v == false }
+    elsif !args.nil? && (args.is_a? Class)
+      my_each { |v| return false if v.class != args }
+    elsif !args.nil? && args.instance_of?(Regexp)
+      my_each { |v| return false unless args.match(v) }
+    else
+      my_each { |v| return false if v != args }
+    end
+    true
+  end
```
Comment (mahmoud717) [inline main.rb]: - my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric

File: main.rb [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+      my_each { |v| return true if v.nil? || v == true }
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each { |v| return true if v.instance_of?(arg) }
+    elsif !arg.nil? && arg.instance_of?(Regexp)
+      my_each { |v| return true if arg.match(v) }
+    else
+      my_each { |v| return true if v == arg }
+    end
+    false
+  end
```
Comment (mahmoud717) [inline main.rb]: - my_any? when no block or argument is given returns false if at least one of the collection is not true
- my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric

File: main.rb [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+      end
+
+      if arg.instance_of?(Regexp)
+        my_each { |i| return false if arg.match(i) }
+        return true
+      end
+
+      my_each { |i| return false if i == arg }
+      return true
+    end
```
Comment (mahmoud717) [inline main.rb]: - my_none? when no block or argument is given returns true only if none of the collection members is true
- my_none? when no block or argument is given returns false only if one of the collection members is true

File: main.rb [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+      accumulator
+    elsif !sym.nil? && (sym.is_a?(Symbol) || sym.is_a?(String))
+      accumulator = num
+      my_each do |item|
+        accumulator = accumulator.nil? ? item : accumulator.send(sym, item)
+      end
+      accumulator
+    end
+  end
+end
```
Comment (mahmoud717) [inline main.rb]: - my_inject raises a "LocalJumpError" when no block or argument is given

**Review by mahmoud717 — CHANGES_REQUESTED** [review]
### Changes required ♻️ 
**Hi @Davidosky007 & @uchennaanya**
Great job with the project so far, but, there are still some changes to be made.
---
### Requested 🔴 
- please move all the test code from the main file into a test file
- please check the inline comments.
---
please make all the changes requested and file for another code review.

**Review by ElijahTobs — APPROVED** [review]
# Status: Approved! :tada: 

## Congratulations! @Davidosky007 and @uchennaanya  

 Good job implementing the changes requested by the previous TSE.

### Before you merge and submit, make sure you work on the link to the live demo in your README.md file. In this case, you're expected to have a link to an online IDE like [repl](https://repl.it) that contains your code.

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Pull request merged**